gapic-common/lib/gapic/common/retry_policy.rb

```diff
@@ -0,0 +1,196 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gapic/common/error_codes"
+
+module Gapic
+  module Common
+    # Gapic Common retry policy base class.
+    class RetryPolicy
+      DEFAULT_INITIAL_DELAY = 1
+      DEFAULT_MAX_DELAY = 15
+      DEFAULT_MULTIPLIER = 1.3
+      DEFAULT_RETRY_CODES = [].freeze
+      DEFAULT_TIMEOUT = 3600 # One hour
+
+      ##
+      # Create new Gapic::Common:RetryPolicy instance.
+      #
+      # @param initial_delay [Numeric] Initial delay in seconds for this policy.
+      # @param max_delay [Numeric] Maximum delay in seconds for this policy.
+      # @param multiplier [Numeric] The delay scaling factor for each subsequent retry attempt.
+      # @param retry_codes [Array<String|Numeric>] List of retry codes.
+      # @param timeout [Numeric] Timeout threshold value in seconds.
+      #
+      def initialize initial_delay: nil, max_delay: nil, multiplier: nil, retry_codes: nil, timeout: nil
+        instance_variable_set "@initial_delay", initial_delay
```

Comment:
Instead of using `instance_variable_set`, just

```ruby
@initial_delay = initial_delay
```

Reply:
Additionally, it's probably better to do the defaulting here rather than in the getter methods below. That is:

```ruby
@initial_delay = initial_delay || DEFAULT_INITIAL_DELAY
```

Then below you can simply

```ruby
attr_reader :initial_delay
```

Reply:
`Gapic::CallOptions::RetryPolicy` has a method `apply_defaults`, which checks the instance variables to determine if things were changed from "default values" (i.e. `nil` if they were not). Early defaulting in the initialization would break this behavior.

Reply:
Oh, good point. I hadn't considered that. 👍🏼